conf/gulp-tasks/template/precompiletemplatestaskfactory.js

```diff
@@ -0,0 +1,44 @@
+const TranslateHelperVisitor = require('../../i18n/translatehelpervisitor');
+
+const precompileTemplates = require('./precompiletemplates');
+const { addLocalePrefix, getPrecompiledFileName } = require('./filenameutils');
+
+/**
+ * PrecompileTemplatesTaskFactory is a class for generating localized template bundle tasks.
+ */
+class PrecompileTemplatesTaskFactory {
+  constructor (translator, locale) {
+    this._locale = locale;
+    this._translator = translator;
+  }
+
+  /**
+   * Creates the precompileTemplates task for this factory's locale,
+   * and customizes its name.
+   * @returns {Function}
+   */
+  create () {
```

Comment:
Would it be simpler to just have a method that people could invoke directly (a la bundletemplates.js) and get rid of this factory? One argument against having a factory for this is it only builds a single thing. It can't provide you with different kinds of tasks, just the one. The Factory pattern is typically used when you can build multiple kinds of an object.

Reply:
removed the factory! we shouldn't ever be having different versions of the precompile step, as far as I can tell anyways, so the factory is pretty pointless